lib/url.js

```diff
@@ -442,98 +442,39 @@ function validateHostname(self, rest, hostname) {
 // Automatically escape all delimiters and unwise characters from RFC 2396.
 // Also escape single quotes in case of an XSS attack.
 // Return the escaped string.
+const escapedCharacterCodes = {
+  9: '%09', 10: '%0A', 13: '%0D', 32: '%20',
+  34: '%22', 39: '%27', 60: '%3C', 62: '%3E',
+  92: '%5C', 94: '%5E', 96: '%60', 123: '%7B',
+  124: '%7C', 125: '%7D'
+};
 function autoEscapeStr(rest) {
   var escaped = '';
   var lastEscapedPos = 0;
   for (var i = 0; i < rest.length; ++i) {
     // Manual switching is faster than using a Map/Object.
```

Comment:
Isn't the whole point of having these switch statements to avoid the overhead of accessing a `Map`/object? (Although that looks quite Crankshaftscript-ish)...if we are going to access an object anyway, then the switch statements can just be replaced with something like `const code = escapedCharacterCodes[rest.charCodeAt(i)]; if (code)...`, and this comment can be removed.

Reply:
That's exactly what I do in the [first commit](https://github.com/nodejs/node/pull/18613/commits/e1eb30a4ec99d1178533ba02be01a6c874845558), but it causes ~10% performance regression in `url.parse`.

Reply:
@bmeurer Would you be interested in taking a look at this?

Reply:
Intuitively I'd suggest to go with an array instead of an object literal. Did you try with an array?